warehouse/macaroons/auth_policy.py

```diff
@@ -0,0 +1,120 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pyramid.authentication import CallbackAuthenticationPolicy
+from pyramid.interfaces import IAuthenticationPolicy, IAuthorizationPolicy
+from pyramid.security import Denied
+from pyramid.threadlocal import get_current_request
+from zope.interface import implementer
+
+from warehouse.cache.http import add_vary_callback
+from warehouse.macaroons.interfaces import IMacaroonService
+from warehouse.macaroons.services import InvalidMacaroon
+
+
+def extract_http_macaroon(request):
+    """
+    A helper function for the extraction of HTTP Macaroon from a given request.
+    Returns either a ``None`` if no macaroon could be found, or the byte string
+    that represents our Macaroon.
+    """
+    authorization = request.headers.get("Authorization")
+    if not authorization:
+        return None
+
+    try:
+        auth_method, auth = authorization.split(" ", 1)
+    except ValueError:
+        return None
+
+    if auth_method.lower() != "macaroon":
```

Comment:
I'd expect to see `Authorization: Bearer ABCD` and another layer dispatch on the format of the token.
Is there a spec for new credential type?